src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -198,13 +188,96 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
   },
 }
 
+interface LabelSignalFields {
+  increasedInterest: boolean
+  curatorsPick: boolean
+  // only a partial partner offer type
+  partnerOffer: { end_at: string; active: true } | null
+  primaryLabel: "PARTNER_OFFER" | "INCREASED_INTEREST" | "CURATORS_PICK" | null
+}
+
+// Single function to resolve mutually-exclusive label signals
+const getLabelSignalFields = async (
+  artwork,
+  ctx
+): Promise<LabelSignalFields> => {
+  const signals = {
+    increasedInterest: false,
+    curatorsPick: false,
+    partnerOffer: null,
+    primaryLabel: null as LabelSignalFields["primaryLabel"],
+  }
+
+  const partnerOfferPromise = getActivePartnerOffer(artwork, ctx)
+  const curatorsPickPromise = getIsCuratorsPick(artwork, ctx)
+
+  const [activePartnerOffer, curatorsPick] = await Promise.all([
+    partnerOfferPromise,
+    curatorsPickPromise,
+  ])
+
+  if (artwork.increased_interest_signal) {
+    signals.increasedInterest = true
+  }
```

Comment:
Seems like it'd be much more clear to just return the whole signals object here rather than mutating pieces of it.

Reply:
I went about this two different ways which explains the object here. In one case I assembled one giant signals object to resolve every related fields (eg primaryLabel, partnerOffer, curatorsPick and increasedInterest), then used that function in each resolver and accessed the property. In [99edba3](https://github.com/artsy/metaphysics/pull/5956/commits/99edba36def15fc25021cc56cda3862d863b55de) I switched this to have each resolver use only its piece, and the `primaryLabel` to use the same helper functions in its place.

In the former case I wrote it to imperatively update the object because it felt easier to read for future devs, but this isn't really necessary any more. I'll update to just use normal variable assignment and returning a single value before merging.